script/BaseScript.s.sol

```diff
@@ -374,60 +386,59 @@ contract BaseScript is BaseData, Utils {
 
     function createImmutableMultiChainDeployerSalt(address _deployerAddress)
         internal
-        pure
+        view
         returns (bytes32 _salt)
     {
         _salt = createSalt(_deployerAddress, "MultiChainDeployer");
     }
 
     function createL1YnOFTAdapterUpgradeableProxySalt(address _deployerAddress)
         internal
-        pure
+        view
         returns (bytes32 _salt)
     {
         _salt = createSalt(_deployerAddress, "L1YnOFTAdapterProxy");
     }
 
-    function createL1YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL1YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L1YnOFTAdapter");
     }
 
     function createL2YnOFTAdapterUpgradeableProxySalt(address _deployerAddress)
         internal
-        pure
+        view
         returns (bytes32 _salt)
     {
         _salt = createSalt(_deployerAddress, "L2YnOFTAdapterProxy");
     }
 
-    function createL2YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnOFTAdapter");
     }
 
-    function createL2YnERC20UpgradeableProxySalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnERC20UpgradeableProxySalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnERC20Proxy");
     }
 
-    function createL2YnERC20UpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnERC20UpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnERC20");
     }
 
-    function createL1YnOFTAdapterTimelockSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL1YnOFTAdapterTimelockSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L1YnOFTTimelock");
     }
 
-    function createL2YnOFTAdapterTimelockSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnOFTAdapterTimelockSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
```

Comment:
why is this changed and not kept to pure?

Reply:
The reason for this change is that if the version and label were kept the same, the salt is the same for deployments for different tokens, which created conflicts while deploying the contracts.

Therefore I made the salt dependent on the symbol of the token being deployed.

Reply:
The above meant that it had to be changed to view since it reads from the state (baseInput.erc20Symbol)